l10n_br_sale/models/sale.py

```diff
@@ -0,0 +1,384 @@
+        default=0.00,
+        digits=dp.get_precision("Account"),
+        readonly=True,
+        states={"draft": [("readonly", False)]})
+
+    amount_costs = fields.Float(
+        compute="_compute_costs_value",
+        inverse="_inverse_amount_costs",
+        string="Other Costs",
+        default=0.00,
+        digits=dp.get_precision("Account"),
+        readonly=True,
+        states={"draft": [("readonly", False)]})
+
+    amount_insurance = fields.Float(
+        compute="_compute_costs_value",
+        inverse="_inverse_amount_insurance",
+        string="Insurance",
+        default=0.00,
+        digits=dp.get_precision("Account"),
+        readonly=True,
+        states={"draft": [("readonly", False)]})
+
+    amount_total = fields.Float(
+        compute="_amount_all",
+        string="Total",
+        store=True,
+        digits=dp.get_precision("Account"),
+        help="The total amount.")
+
+    @api.onchange('discount_rate')
+    def onchange_discount_rate(self):
+        for sale_order in self:
+            for sale_line in sale_order.order_line:
+                sale_line.discount = sale_order.discount_rate
+
+    @api.onchange("operation_id")
+    def _onchange_operation_id(self):
+        self.fiscal_position_id = self.operation_id.fiscal_position_id
+
+    # TODO FIscal Comment
+    # @api.model
+    # def _fiscal_comment(self, order):
+    #     fp_comment = []
+    #     fp_ids = []
+    #
+    #     for line in order.order_line:
+    #         if line.operation_line_id and \
+    #                 line.operation_line_id.inv_copy_note and \
+    #                 line.operation_line_id.note:
+    #             if line.operation_line_id.id not in fp_ids:
+    #                 fp_comment.append(line.operation_line_id.note)
+    #                 fp_ids.append(line.operation_line_id.id)
+    #
+    #      return fp_comment
+
+    @api.multi
+    def _prepare_invoice(self):
+        self.ensure_one()
+        result = super(SaleOrder, self)._prepare_invoice()
+
+        if self.operation_id:
+            result['operation_id'] = self.operation_id.id
+            result['fiscal_document_id'] = False
+            result['operation_type'] = self.operation_id.operation_type
+            result['partner_shipping_id'] = self.partner_shipping_id.id
+
+            # TODO Defini document_type_id in other method in line
+            result['document_type_id'] = self._context.get('document_type_id')
+
+            if self.operation_id.journal_id:
+                result['journal_id'] = self.operation_id.journal_id.id
+
+        comment = []
+        if self.note and self.copy_note:
+            comment.append(self.note)
+
+        # TODO FISCAL Commnet
+        # result['fiscal_comment'] = " - ".join(fiscal_comment)
+        # fiscal_comment = self._fiscal_comment(self)
+        result['comment'] = " - ".join(comment)
+        return result
+
+    @api.multi
+    def action_invoice_create(self, grouped=False, final=False):
```

Comment:
hum overriding this method without calling super is quite annoying... I understand the why (separating NFSe) but it can create conflicts with other modules. From a quit search

- in the official addons it will only conflict with the **repair** addons.
- in the OCA, it will at least conflict with these modules in OCA/sale-workflow: **sale_invoice_plan**, **sale_triple_discount**, **sale_automatic_workflow**

Reply:
É importante corrigir isso!

Reply:
Hi @rvalyi one way to solve it are call super and after searching in created invoice the line with service delete and create a new one, a better solution are suggest PR in the core to make similar things as @mileo propused in stock_picking_invoicing